external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -456,43 +464,101 @@ impl<'env> Context<'env> {
         }
     }
 
-    fn enter_nominal_block(&mut self, name: Option<P::BlockLabel>, name_type: NominalBlockType) {
+    fn enter_nominal_block(
+        &mut self,
+        loc: Loc,
+        name: Option<P::BlockLabel>,
+        name_type: NominalBlockType,
+    ) {
         debug_assert!(
             self.nominal_blocks.len() < 100,
             "Nominal block list exceeded 100."
         );
-        let sym = if let Some(name) = name {
-            name.value()
-        } else {
-            // all named blocks have names, so a non-named block must be a loop
-            N::Exp_::LOOP_NAME_SYMBOL
-        };
         let id = self.nominal_block_id;
         self.nominal_block_id += 1;
-        self.nominal_blocks.push((sym, id, name_type));
+        self.nominal_blocks
+            .push((loc, name.map(|n| n.value()), id, name_type));
```

Comment:
To match the above changes:

```suggestion
        let block_label = generate_block_label(loc, name, id, name_type);
        self.nominal_blocks
            .push((name.map(|n| n.value()), id, name_type));
```

with:

```
fn generate_block_label(
    loc: Loc,
    name: Option<Symbol>,
    id: u16,
    name_type: NominalBlockType,
) -> BlockLabel {
    use NominalBlockType as NBT;
    let is_implicit = name.is_none();
    let name = match name_type {
      NBT::Loop(_) => name.unwrap_or(BlockLabel::IMPLICIT_LABEL_SYMBOL);
      NBT::Block => name.expect("ICE can't arrive here"),
      NBT::LambdaBody | NBT::LambdaBreak => {
          assert(name.is_none(), "ICE lambda body is labeled");
          BlockLabel::IMPLICIT_LABEL_SYMBOL
      }
    };
    let var_ = N::Var_ { name, id, color: 0 };
    let label = sp(loc, var_);
    BlockLabel { label, is_implicit }
}
```